core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -398,9 +401,33 @@ class HttpGoogleServicesDAO(
 
   override def getBucket(bucketName: String)(implicit executionContext: ExecutionContext): Future[Option[Bucket]] = {
     implicit val service = GoogleInstrumentedService.Storage
-    val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
-    retryWithRecoverWhen500orGoogleError(() => { Option(executeGoogleRequest(getter)) }) {
-      case e: HttpResponseException => None
+    retryWithRecoverWhen500orGoogleError(() => {
+      val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
+      Option(executeGoogleRequest(getter))
+    }) {
+      case _: HttpResponseException => None
+    }
+  }
+
+  override def getRegionForRegionalBucket(bucketName: String): Future[Option[String]] = {
+    getBucket(bucketName) map { maybeBucket =>
+      val bucket = maybeBucket.getOrElse(throw new RawlsException(s"Failed to retrieve bucket `$bucketName`"))
+      bucket.getLocationType match {
+        case SingleRegionLocationType => Option(bucket.getLocation.toLowerCase)
```

Comment:
Is there a reason for the `toLowerCase` here?

Reply:
Yes. The reason we convert it to lowercase is that `.regions().get(..)` in [here](https://github.com/broadinstitute/rawls/blob/889918b9e49916d7ac9832539854f89b4b033e3f/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala#L425-L426) expects the input to be in lowercase. If it is in any other format it throws an error

Reply:
Maybe comment that here?

Reply:
So I refactored the code a bit. Instead of doing `.toLowerCase` here, I shifted it to `getComputeZonesForRegion()` function where it made more sense. I have also added a comment there stating why `.toLowerCase` is needed